src/ucs/datastruct/arbiter.c

```diff
@@ -158,8 +155,8 @@ void ucs_arbiter_group_purge(ucs_arbiter_t *arbiter,
             prev->next = next;
         } else {
             /* keep the element */
-            ptr->next = next; /* Restore next pointer */
-            prev      = ptr;
+            ucs_arbiter_elem_set_scheduled(ptr, group);
+            prev       = ptr;
```

Comment:
why don't need to restore ptr->next?

Reply:
because now we reset ->group instead of ->next - added comment